h/views/account_signup.py

```diff
@@ -257,6 +324,59 @@ def post(self):
             headers=login(user, self.request),
         )
 
+    @exception_view_config(
+        context=EmailConflictError,
+        route_name="signup.orcid",
+        renderer="h:templates/error.html.jinja2",
+    )
+    @exception_view_config(
+        context=EmailConflictError,
+        route_name="signup.google",
+        renderer="h:templates/error.html.jinja2",
+    )
+    @exception_view_config(
+        context=EmailConflictError,
+        route_name="signup.facebook",
+        renderer="h:templates/error.html.jinja2",
+    )
+    def email_conflict_error(self):
+        self.request.session.flash(
+            f"There's already a Hypothesis account with your {self.settings.provider_name} email address."
+            " Try logging in or resetting your password."
+            f" Once logged in you can connect {self.settings.provider_name} in your account settings.",
+            "error",
+        )
+
+        return HTTPFound(
+            self.request.route_url(
+                "login", _query={"username": self.decode_idinfo().email}
+            ),
+            headers=logout(self.request),
+        )
+
+    @exception_view_config(
+        context=IdentityConflictError,
+        route_name="signup.orcid",
+        renderer="h:templates/error.html.jinja2",
+    )
+    @exception_view_config(
+        context=IdentityConflictError,
+        route_name="signup.google",
+        renderer="h:templates/error.html.jinja2",
+    )
+    @exception_view_config(
+        context=IdentityConflictError,
+        route_name="signup.facebook",
+        renderer="h:templates/error.html.jinja2",
+    )
+    def identity_conflict_error(self):
+        self.request.session.flash(
+            f"There's already a Hypothesis account connected to your {self.settings.provider_name} account. Try logging in.",
+            "error",
+        )
+
+        return HTTPFound(self.request.route_url("login"), headers=logout(self.request))
```

Comment:
Same deal: this is called if either a `GET` or a `POST` to `/signup/<PROVIDER>` raises `IdentityConflictError`.